samples/system-test/files.test.js

```diff
@@ -137,13 +141,72 @@ it('should make a file public', async () => {
   );
 });
 
-it('should generate a signed URL for a file', async () => {
+it('should generate a v2 signed URL for a file', async () => {
   const output = await exec(
     `${cmd} generate-signed-url ${bucketName} ${copiedFileName}`
   );
   assert.match(output, new RegExp(`The signed url for ${copiedFileName} is `));
 });
 
+it('should generate a v4 signed URL and read a file', async () => {
+  const output = await exec(
+    `${cmd} generate-v4-read-signed-url ${bucketName} ${copiedFileName}`
+  );
+
+  const regExp = new RegExp(
+    'Generated GET signed URL:\n' +
+      '(.*)\n' +
+      'You can use this URL with any user agent, for example:\n' +
+      'curl '
+  );
+  assert.match(output, regExp);
+
+  const match = output.match(regExp);
+  const res = await fetch(match[1]);
+  const text = await res.text();
+  assert.strictEqual(text, fileContent);
+});
+
+it('should generate a v4 signed URL and upload a file', async () => {
+  const output = await exec(
+    `${cmd} generate-v4-upload-signed-url ${bucketName} ${signedFileName}`
+  );
+  const regExp = new RegExp(
+    'Generated PUT signed URL:\n' +
+      '(.*)\n' +
+      'You can use this URL with any user agent, for example:\n' +
+      'curl -X PUT'
+  );
+  assert.match(output, regExp);
+
+  const match = output.match(regExp);
+  const req = {
+    method: 'PUT',
+    headers: {'Content-Type': 'application/octet-stream'},
+    body: fileContent,
+  };
+  await fetch(match[1], req);
```

Comment:
In the name of consistency, I wonder if we should use https://www.npmjs.com/package/teeny-request?

Reply:
In that case we need to migrate the root system-test as well, they're using fetch right now.

I'll change it here, and open an issue to migrate that as well.

Reply:
Having looked at the code, I think I'd prefer keeping `node-fetch` as we'll likely have more code to use `teeny-request` since it's callback-only.